packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -265,18 +243,24 @@ class _NutritionPageLoadedState extends State<NutritionPageLoaded>
             ),
           ),
         );
+
+        if (_nutrientToHighlight == orderedNutrient) {
+          onNextFrame(() => _focusNodes[i].requestFocus());
+          _nutrientToHighlight = null;
+        }
       }
       children.add(
         NutritionAddNutrientButton(
           nutritionContainer: _nutritionContainer,
-          refreshParent: () => setState(() {}),
+          onNutrientSelected: (final OrderedNutrient nutrient) {
+            setState(() => _nutrientToHighlight = nutrient);
+          },
         ),
       );
-
-      if (_ownerFieldBannerVisible) {
-        children.add(SizedBox(height: _ownerFieldBannerHeight));
-      }
     } else {
+      for (final Nutrient nutrient in _controllers.keys) {
+        _controllers[nutrient]!.dispose();
```

Comment:
Very surprising to use `dispose` here.
At least that would deserve a comment.

Reply:
We clear the list of controllers and create new ones.
The `dispose` method ensures that we won't have a memory leak because some listener(s) may still exist